File: Data Structures & Algorithms/reverse-bits/submission-1.kt
class Solution {
    fun reverseBits(n: Int): Int {
        var result = 0
        for (i in 0 until 32) {
            val current = (n shr i) and 1
            result = result or (current shl (31 - i))
        }
        return result
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        result = []

        def dfs(index: int, current: List[int], total: int):
            if index >= len(nums) or total > target:
                return
            if total == target:
                result.append(current[:])
                return

            dfs(index, current + [nums[index]], total + nums[index])
            dfs(index + 1, current, total)

        dfs(0, [], 0)

        return result